api4/team.go

```diff
@@ -1405,6 +1405,9 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("emails", emailList)
 
 	if len(memberInvite.ChannelIds) > 0 {
+		// Check if the user sending the invitation has access to the channels where the invitation is being sent
+		c.App.ValidateUserPermissionsOnChannels(c.AppContext, *c.AppContext.Session(), memberInvite)
```

Comment:
Another way to do this would be to operate directly against the `(model.MemberInvite).ChannelIds` field with a filter.

```go
memberInvite.ChannelIds = c.App.FilterChannelsIdsByAuthzToManageMembers(c.AppContext, *c.AppContext.Session(), memberInvite.ChannelIds)
```

Then the method neither requires knowledge of—nor mutates—`model.EmailInvite`. You can also remove the getters and setters on `EmailInvite` and `MemberInvite`.

Reply:
for my knowledge (Im still newbie with golang), any particular advantage of doing it with that approach? I choosed the reference one so there is no need to create a copy in memory.

Reply:
One advantage is readability as it was not clear to me that `ValidateUserPermissionsOnChannels` was actually filtering them some channels as well - at first glance I was expecting some type or response of only the validation check

Reply:
yes, that is an advantage!